### PR TITLE
[Core] Remove legacy methods in `ManagerInterfaceFactoryInterface`

### LINKED ISSUE #505: Remove redundant `ManagerInterfaceFactoryInterface` methods
## What

Remove the following methods from `ManagerInterfaceFactoryInterface`:

- `managers`
- `managerRegistered`
- `instantiateUIDelegate`

## Why

They are (aside from the UI method, which is for the future), mostly boiler-plate reduction code, that is not specific to the implementation of the factory, and can/will/are be provided by host-facing middleware such as #507.

### PR DESCRIPTION
**Pull request by foundrytom**
These methods are superseded by functionality in `ManagerFactory`, now `Session` has been removed, these are unused and redundant.

Closes #505

### CODE REVIEW

File: python/openassetio/pluginSystem/PluginSystemManagerInterfaceFactory.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 import os
 
-from ..hostApi.ManagerInterfaceFactoryInterface import ManagerInterfaceFactoryInterface
+from ..hostApi import ManagerInterfaceFactoryInterface
```
Comment (feltech) [inline python/openassetio/pluginSystem/PluginSystemManagerInterfaceFactory.py:24]: [consider] Should we also remove the `managers` and `managerRegistered` methods from the `PluginSystemManagerInterfaceFactory`?

**Review by feltech — APPROVED** [review]

**Comment by feltech** [comment]
[fix] Noticed whilst reviewing #536 - there is a TODO in `test_managerfactory.py:285`
```py
    # TODO(DF): @pylint Remove once ManagerImplementationFactoryInterface
    #   fully migrated to C++.
    # pylint: disable=abstract-method
```

**Comment by foundrytom** [comment]
> [fix] Noticed whilst reviewing #536 - there is a TODO in `test_managerfactory.py:285`

Good spot, thanks, removed [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/535/commits/308f4c4d0234f09fe196d7c3602b63c6b6d1c6a3).

File: python/openassetio/pluginSystem/PluginSystemManagerInterfaceFactory.py (hunk 1 above)
Comment (foundrytom) [reply python/openassetio/pluginSystem/PluginSystemManagerInterfaceFactory.py:24]: I knew I forgot something! Removed [here], thanks.

Comment (foundrytom) [reply python/openassetio/pluginSystem/PluginSystemManagerInterfaceFactory.py:24]: Which also allowed me to notices [this](https://github.com/OpenAssetIO/OpenAssetIO/pull/535/commits/139ed3462b44150da49cf857f39940985f2faf92).

File: python/openassetio/pluginSystem/PluginSystemManagerInterfaceFactory.py [hunk 2]
```diff
@@ -88,65 +85,8 @@ def identifiers(self):
...
-
-    def managerRegistered(self, identifier):
-        """
-        @return bool, True if the supplied identifier is known to the
-        factory.
-        """
-        if not self.__pluginManager:
-            self.__scan()
-
-        return identifier in self.__pluginManager.identifiers()
```
Comment (feltech) [inline python/openassetio/pluginSystem/PluginSystemManagerInterfaceFactory.py:88]: [consider] pep8 too many blank lines
```suggestion
```

**Review by feltech — APPROVED** [review]
Just a minor PEP8 thing, otherwise happy for this to be squashed and merged.

File: python/openassetio/pluginSystem/PluginSystemManagerInterfaceFactory.py (hunk 2 above)
Comment (foundrytom) [reply python/openassetio/pluginSystem/PluginSystemManagerInterfaceFactory.py]: Fixed in rebase

**Pull request merged**